### ISSUE TITLE
Deprecate the `runPythonAsync` API?

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#1133]
We are getting close to asyncio support. #876 and #1041 added implementations of `eval_code_async` and #1109 adds tests to `eval_code_async` to show that it works (of course #1109 isn't merged yet but it's just testing logic that has been merged). #880 added the ability to await javascript Promises. The only ingredients still missing are an event loop implementation (@oeway is working on this in PR #958) and the inverse conversion Python future ==> Js promise so that one can await a Python future from javascript.

So I think we need to consider what to do with the `runPythonAsync` API. I think we can deprecate `runPythonAsync` in favor of `loadPackagesFromImports` and `runPython` separately: this is anyways more flexible because instead of using `runPython` one can directly use `eval_code` which is quite parametrizable now thanks to @casatir. (Also, I found `runPythonAsync` to be a very confusing name when I first started using pyodide, I assumed it had something to do with `asyncio` but when I realized it didn't I was just really not sure what it did.)

I guess instead of making an api `runPythonActualAsync` that uses `eval_code_async`, we can just have people use `pyodide_py.eval_code_async` directly to avoid name clashes with the deprecated API.

@casatir @dalcde @oeway @rth Thoughts?

**Comment by casatir** [comment]
Thanks @hoodmane for this crystal clear exposition of the async situation and the motivation behind this issue.

I totally agree that `runPythonAsync/runPython` is puzzling. It was hard to me to understand that the former actually loads Pyodide packages while the later doesn't. It seems to me that having `runPython` that does not actually offers all the power of Pyodide is useless. A nice name like this should be the entry point for great things.

That said, the API that I have in mind is not exactly the one presented by @hoodmane. I am more thinking to a parametrized `runPython`where the default behavior is `loadPackagesFromImports + eval_code`. Something like

```javascript
async function runPython(code, loadPackages=true, pythonAsync=false) {
    if( loadPackages ) {
        await loadPackagesFromImports(code);
    }
    if( pythonAsync ) {
        return await pyodide_py.eval_code_async(code);
    } else {
        return pyodide_py.eval_code(code);
    }
}
```

This will of course break the current API.

If it is really important to have `runPython(code, false, false)` returning directly the result (not async), it is possible to remove async/await and use only promise chaining.

Another thought on this. Why the Python API is not entitled to have the package loaded from imports? That being said, we can have a new `eval_code` that preload packages (even without the async support in Pyodide):

```python
def eval_code_with_package_preload(code, *args, **kwargs):
    return js.pyodide.loadPackagesFromImports(source).then(lambda *_: eval_code(code, *args, **kwargs))
```

The previous `runPython` will be only be a JS wrapper to `eval_code_with_package_preload`. Note that this could become our `eval_code`.

**Comment by hoodmane** [comment]
Closed in favor of #1496.
